### DISCUSSION TITLE
Security For Supabase Realtime?

### DISCUSSION THREAD

**Discussion by HoyaBoya** [supabase/supabase#457] [Questions]
# TLDR

What's the point of `SECRET_KEY_BASE` when starting up a Supabase Realtime Server when I can connect through the JS client with no credentials?

UPDATE: Never mind, I finally caught this at the bottom while searching through the code. There is no security. It would be helpful if this were on the main README.

https://github.com/supabase/realtime/blob/36690138eb9c33258ec6f3d1890e7232e6e70d64/examples/next-js/README.md#note

# Long Story

I'm doing an evaluation of Supabase Realtime for our company and everything seems to work fine in the Getting Started instructions on my local development machine using the docker image. My only concern is that I'm able to connect with a JS client without any credentials. I'd expect a 401.

I went through all the examples and documentation and I can't find prior art for how this should work. 

## JS Client Example

```
const { RealtimeClient } = require('@supabase/realtime-js')

var client = new RealtimeClient('https://127.0.0.1:4000/socket')

const run = async () => {
  await new Promise((resolve, reject) => {
    client.onOpen(() => console.log('Socket opened.'))
    client.onClose(() => console.log('Socket closed.'))
    client.onError(e => console.log('Socket error', e.message))

    console.log('Connecting...')
    client.connect()
    console.log('Connected!')

    // Listen to events on the entire database.
    var databaseChanges = client.channel('realtime:*')
    databaseChanges.on('*', e => console.log(e))
    databaseChanges.on('INSERT', e => console.log(e))
    databaseChanges.on('UPDATE', e => console.log(e))
    databaseChanges.on('DELETE', e => console.log(e))
    databaseChanges.subscribe()
  })
}

run()
  .then(() => {
    process.exit(0)
  })
  .catch(e => {
    console.error(e)
    process.exit(1)
  })
```

**Comment by kiwicopple** [comment]
Hey @HoyaBoya - sorry for the slow response. 

We're working on security here: https://github.com/supabase/realtime/pull/99

**Comment by w3b6x9** [comment]
@HoyaBoya we now have Realtime security based on RLS policies introduced towards the end of last year. You can read more about it here: https://supabase.com/blog/realtime-row-level-security-in-postgresql.
